pkg/engine/resolve/resolve.go

```diff
@@ -174,24 +191,27 @@ func (c *Context) Clone() Context {
 		copy(patches[i].extraPath, c.patches[i].extraPath)
 		copy(patches[i].data, c.patches[i].data)
 	}
+	undefinedVariables := make([]string, len(c.undefinedVariables))
+	copy(undefinedVariables, c.undefinedVariables)
 	return Context{
-		Context:         c.Context,
-		Variables:       variables,
-		Request:         c.Request,
-		pathElements:    pathElements,
-		patches:         patches,
-		usedBuffers:     make([]*bytes.Buffer, 0, 48),
-		currentPatch:    c.currentPatch,
-		maxPatch:        c.maxPatch,
-		pathPrefix:      pathPrefix,
-		beforeFetchHook: c.beforeFetchHook,
-		afterFetchHook:  c.afterFetchHook,
-		position:        c.position,
+		ctx:                c.ctx,
```

Comment:
Should we actually allow this? Might be another possible memory leak.
Wouldn't it be better if we change it to `Clone(ctx context.Context)` to force the user to explicitly set the context?
This way, they could create a new context with the content they want and pass it on.
Alternatively, we could not clone context.Context at all, leave Clone() as is, and provide CloneWithContext(ctx context.Context). WDYT?

Reply:
Oh, good idea. Maybe we can even entirely remove `Clone()` in favour of `WithContext()` let me look into it and get back to you in a bit

Reply:
The only usage of `.Clone()` in the whole package is in https://github.com/wundergraph/graphql-go-tools/blob/f29b89066be95f556ef7acdc1e67ba027d3a6d5f/pkg/engine/resolve/resolve.go#L826 - which looks fine to me since the contexts are cloned, executed in parallel and then we wait for all to finish before returning. However, this API is so easy to misuse that I would make it private by renaming it to `.clone()` and I would also add a comment to the method clarifying that it needs to be used with care. WDYT?

Reply:
I've renamed `.Clone()` to `.clone()`